### PR TITLE
Support for Default Value

### PR DESCRIPTION
**Pull request by jagadeeshym**

### CODE REVIEW

File: src/components/ColorPalleteComponent.tsx [hunk 1]
```diff
@@ -32,12 +32,12 @@ export class ColorPalleteComponent extends React.Component<InlineComponentProps,
...
-        const g = parseInt(hex.substring(2,4), 16);
-        const b = parseInt(hex.substring(4,6), 16);
-        return 'rgba('+r+','+g+','+b+','+opacity/100+')';
+    convertHex(hex: string, opacity: number) {
+        hex = hex.replace("#", "")
+        const r = parseInt(hex.substring(0, 2), 16)
+        const g = parseInt(hex.substring(2, 4), 16)
+        const b = parseInt(hex.substring(4, 6), 16)
+        return `rgba(${r}, ${g}, ${b}, ${opacity / 100})`
     }
```
Comment (jagadeeshym) [inline src/components/ColorPalleteComponent.tsx]: Fixing lint errors

File: src/components/IterableEditComponent.tsx [hunk 2]
```diff
@@ -495,11 +497,7 @@ export class IterableEditComponent extends React.Component<ImageUploadProps | It
     }
 
     createNew = () => {
-        if (this.props.field.iterabletype.type === "nested") {
-            this.props.modelChanged(_.concat(this.state.model, {}))
-        } else {
-            this.props.modelChanged(_.concat(this.state.model, ""))
-        }
+        this.props.modelChanged(_.concat(this.state.model, this.getIterableDefaultValue(this.props.field.iterabletype)))
     }
```
Comment (jagadeeshym) [inline src/components/IterableEditComponent.tsx]: New Iterable Field will be setted with default Values. Which is given in config

File: src/components/IterableEditComponent.tsx [hunk 3]
```diff
@@ -524,12 +522,27 @@ export class IterableEditComponent extends React.Component<ImageUploadProps | It
 
     addAtIndex = (index: any) => {
         const clone = _.cloneDeep(this.state.model)
-        if (this.props.field.iterabletype.type === "nested") {
-            clone.splice(index, 0, {});
+        clone.splice(index, 0, this.getIterableDefaultValue(this.props.field.iterabletype))
+        this.props.modelChanged(clone)
+    }
+
```
Comment (jagadeeshym) [inline src/components/IterableEditComponent.tsx]: This method will return corresponding default value, based on iterableType. Object for nested type. string or boolean for others

File: src/components/NestedEditComponent.tsx [hunk 4]
```diff
@@ -245,6 +247,26 @@ export class NestedEditComponent extends React.Component<InlineComponentProps, a
...
+    }
+
+    componentDidMount() {
+        this.updateDefaultValue(this.props)
+    }
+
+    componentWillReceiveProps(nextProps: any) {
+        this.updateDefaultValue(nextProps)
+    }
+
```
Comment (jagadeeshym) [inline src/components/NestedEditComponent.tsx]: Updating default value for each nested component taken from config

File: readme.md [hunk 5]
```diff
@@ -427,6 +427,34 @@ If the field is itself an object containing more fields, its _type_ should be "n
...
+This is to support default values for our components. In each config, we can represent a defaultValue key. Custom Function will be called, when the
+component does not have value.
+
+```
+{
+    "title": "Text",
+    "editable": true,
+    "display": true,
+    "type": "text",
+    "defaultValue": () => "Initial Default Value"
```
Comment (kunal15595) [inline readme.md]: call it defaultValueFn or defaultValueEvaluator, latter is better

Comment (jagadeeshym) [reply readme.md]: updated to defaultValueFn

**Pull request merged**